hooks/webs/useWebDetails.tsx

```diff
@@ -0,0 +1,28 @@
+import type { Location } from '@prisma/client'
+import { REMOTE_URL } from '@helpers/config'
+import { useQuery } from '@tanstack/react-query'
+
+export async function fetchWebDetailsRequest(isSSR = false) {
+  const response = await fetch(
+    isSSR ? `${REMOTE_URL}/api/web-details` : '/api/web-details',
+  )
+  const data: { webDetails: Location[] } = await response.json()
+  const { webDetails } = data
+  return webDetails || []
+}
+
+export default function useWebDetails() {
+  const {
+    data: webDetails,
+    isLoading,
+    isError,
+  } = useQuery(['webDetails'], () => fetchWebDetailsRequest(false), {
+    refetchOnWindowFocus: false,
+  })
+
+  return {
+    webDetails,
+    isLoading,
+    isError,
+  }
+}

```

Comment:
There is already a `useWebs` and an associated API route though, do you think we can make use of those?

Reply:
So that's what I started with, however, it pulls more data, rather than just a scan on the Location table

Reply:
Updated, this is now `useWebs` and the old one is now `useWebListings`, please test anything that used the old one as it is in an area I'm unfamiliar with, including the admin areas 😱.